code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -41,124 +50,136 @@
 			if(do_after(user, src, 25))
-				user.visible_message("[user] removes the fill cap on \the [src].","<span class='notice'>You remove the fill cap on \the [src].</span>")
+				user.visible_message("<span class='notice'>[user] removes the fill cap on [src].</span>",
+				"<span class='notice'>You remove the fill cap on [src].</span>")
 				playsound(src,'sound/items/Ratchet.ogg', 100, 1)
 				flags |= OPENCONTAINER
 		else
-			user.visible_message("[user] begins to seal the fill cap on \the [src].","<span class='notice'>You begin to seal the fill cap on \the [src].</span>")
+			user.visible_message("<span class='notice'>[user] begins to seal the fill cap on [src].</span>",
+			"<span class='notice'>You begin to seal the fill cap on [src].</span>")
 			if(do_after(user, src, 25))
-				user.visible_message("[user] fastens the fill cap on \the [src].","<span class='notice'>You fasten the fill cap on \the [src].</span>")
+				user.visible_message("<span class='notice'>[user] fastens the fill cap on [src].</span>",
+				"<span class='notice'>You fasten the fill cap on [src].</span>")
 				playsound(src,'sound/items/Ratchet.ogg', 100, 1)
 				flags &= ~OPENCONTAINER
 		return
 
-	if (istype(W, /obj/item) && !is_open_container() && !istype(src, /obj/item/weapon/extinguisher/foam) && !istype(W, /obj/item/weapon/storage/evidencebag))
+	if(istype(W, /obj/item) && !is_open_container() && !istype(W, /obj/item/weapon/storage/evidencebag))
 		if(W.is_open_container())
 			return //We're probably trying to fill it
+		if(!can_cram_item)
+			to_chat(user, "<span class='warning'>[src] can't possibly hold and fire any item!</span>")
+			return
 		if(W.w_class > W_CLASS_TINY)
-			to_chat(user, "\The [W] won't fit into the nozzle!")
+			to_chat(user, "<span class='warning'>[W] won't fit into the nozzle!</span>")
 			return
 		if(locate(/obj) in src)
-			to_chat(user, "There's already something crammed into the nozzle.")
+			to_chat(user, "<span class='warning'>There's already something crammed into the nozzle.</span>")
 			return
-		if(isrobot(user) && !isMoMMI(user)) // MoMMI's can but borgs can't
-			to_chat(user, "You're a robot. No.")
+		if(isrobot(user) && !isMoMMI(user)) //MoMMI's can but borgs can't
+			to_chat(user, "<span class='warning'>You're a robot. No.</span>")
 			return
 		if(user.drop_item(W, src))
-			to_chat(user, "You cram \the [W] into the nozzle of \the [src].")
+			to_chat(user, "<span class='notice'>You cram [W] into [src]'s nozzle.</span>")
 			message_admins("[user]/[user.ckey] has crammed \a [W] into a [src].")
 
 /obj/item/weapon/extinguisher/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	if(proximity_flag)
-		if((istype(target, /obj/structure/reagent_dispensers)))
-			target.reagents.trans_to(src, 50, log_transfer = TRUE, whodunnit = user)
-			to_chat(user, "<span class='notice'>\The [src] is now refilled</span>")
+		if(istype(target, /obj/structure/reagent_dispensers/watertank))
```

Comment:
Doesn't this mean extinguishers can no longer be filled with welding fuel from welding tanks?
Why?

Reply:
Snap, correcting that